File: backend/shared/uidl/src/main/java/io/mateu/uidl/data/StatusItem.java
package io.mateu.uidl.data;

import lombok.Builder;

/**
 * One row of a {@link StatusList}: an {@code icon} (or an {@code avatar} — a short text like a
 * person's initials rendered inside a circular avatar, taking precedence over the icon), a {@code
 * title} plus muted {@code description}, and on the right an optional status chip ({@code status}
 * text with a badge-palette {@code statusColor}) and/or up to three small action buttons ({@code
 * actionLabel} + {@code actionId}, and optionally {@code actionLabel2}/{@code actionId2} and {@code
 * actionLabel3}/{@code actionId3} — e.g. "scan document" / "fill in manually" / "no show" on a
 * guest row), each dispatched with {@code { "_item": id }}.
 */
@Builder
public record StatusItem(
    String id,
    String icon,
    String avatar,
    String title,
    String description,
    String status,
    String statusColor,
    String actionLabel,
    String actionId,
    String actionLabel2,
    String actionId2,
    String actionLabel3,
    String actionId3) {}

File: frontend/web/monorepo/libs/mateu/src/mateu/shared/apiClients/dtos/componentmetadata/StatusItem.ts
export default interface StatusItem {
    id?: string
    icon?: string
    avatar?: string
    title?: string
    description?: string
    status?: string
    statusColor?: string
    actionLabel?: string
    actionId?: string
    actionLabel2?: string
    actionId2?: string
    actionLabel3?: string
    actionId3?: string
    /** optional icon names (e.g. "vaadin:pencil") — renderers may show icon-only buttons with the label as tooltip */
    actionIcon?: string
    actionIcon2?: string
    actionIcon3?: string
    /** timeline-style lines under the description ("27 jul 09:15 - comentario") */
    lines?: string[]
}
